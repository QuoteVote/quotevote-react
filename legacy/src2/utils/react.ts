import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 