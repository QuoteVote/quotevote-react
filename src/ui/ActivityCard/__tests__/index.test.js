import React from 'react'
import { render } from '@testing-library/react'

import { ActivityCard } from '..'

describe('<ActivityCard  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<ActivityCard />)
    expect(loadingIndicator.container.firstChild).toMatchSnapshot()
  })
})
