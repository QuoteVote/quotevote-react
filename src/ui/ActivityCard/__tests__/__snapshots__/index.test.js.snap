// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityCard  /> should match snapshot 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-1 makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <div
      class="MuiBox-root MuiBox-root-7"
    >
      <div
        class="makeStyles-activityHeader-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Username
        </h6>
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
        >
          Today @ 3:35PM
        </span>
      </div>
      <p
        class="MuiTypography-root makeStyles-activityBody-3 MuiTypography-body1"
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-9"
  >
    <span
      class="MuiTypography-root MuiTypography-caption"
    >
      +0 / -0
    </span>
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      style="padding: 0px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root fas fa-heart"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
